### PR TITLE
Tanium solution 2.0.1

### PR DESCRIPTION
**Pull request by sdball**
Required items, please complete
   
   Change(s):
   - Add playbooks that integrate with Tanium Patch
   - Fix playbooks POSTing graphql variables
   - Do not create an integration account per playbook

   Reason for Change(s):
   - Deepen the Sentinel integration
   - Update the playbooks to work with the latest versions of the Tanium API Gateway
   - Improve the user experience deploying this integration

   Version Updated:
   - Yes

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

File: Solutions/Tanium/Package/createUiDefinition.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
...
+        "description": "<img src=\"https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Logos/Tanium.svg\" width=\"75px\" height=\"75px\">\n\n**Note:** _There may be [known issues](https://aka.ms/sentinelsolutionsknownissues) pertaining to this Solution, please refer to them before installing._\n\nThe [Tanium](https://www.tanium.com/) solution for Microsoft Sentinel enables you to ingest Tanium Threat Response alerts as Sentinel incidents as well as incorporate Tanium's real-time endpoint data.\n\n**Workbooks:** 1, **Analytic Rules:** 1, **Playbooks:** 10\n\n[Learn more about Microsoft Sentinel](https://aka.ms/azuresentinel) | [Learn more about Solutions](https://aka.ms/azuresentinelsolutionsdoc)",
```
Comment (v-vdixit) [inline Solutions/Tanium/Package/createUiDefinition.json:9]: Hi @sdball please update Sentinel in this line to Microsot Sentinel to resolve branding validation failure, thanks!
image

File: Solutions/Tanium/Package/createUiDefinition.json
Comment (v-vdixit) [inline Solutions/Tanium/Package/createUiDefinition.json:1]: Please update the values of 'Sentinel' to 'Microsoft Sentinel' in line 5661 and line 9543 in this file to fix the branding validation failure, thanks!

image

image

**Comment by v-vdixit** [comment]
@sdball thanks for the updates we are working on this will update you by the end of this week, thanks!

**Comment by v-vdixit** [comment]
Hi @sdball we are checking on the playbook code changes, will update you shortly, thanks!

File: Solutions/Tanium/Playbooks/Tanium-ResolveThreatResponseAlert/azuredeploy.json
Comment (v-vdixit) [inline Solutions/Tanium/Playbooks/Tanium-ResolveThreatResponseAlert/azuredeploy.json:1]: Hi @sdball we have run into a deployment error while creating this playbook, can you please check and update this, 
adding a screenshot of the error for your reference, thanks!

image

File: Solutions/Tanium/Playbooks/Tanium-QuarantineHosts/azuredeploy.json
Comment (v-vdixit) [inline Solutions/Tanium/Playbooks/Tanium-QuarantineHosts/azuredeploy.json:1]: Hi @sdball please add screenshot demonstrating successful working of this playbook, thanks!

File: Solutions/Tanium/Playbooks/Tanium-ComplyFindings/azuredeploy.json
Comment (v-vdixit) [inline Solutions/Tanium/Playbooks/Tanium-ComplyFindings/azuredeploy.json:1]: Hi @sdball please check on this playbook as we are getting the same deployment error as Tanium-ResolveThreatResponseAlert while creating this, please check and update, thanks!

File: Solutions/Tanium/Playbooks/Tanium-QuarantineHosts/azuredeploy.json
Comment (sdball) [reply Solutions/Tanium/Playbooks/Tanium-QuarantineHosts/azuredeploy.json:1]: Hello! We already have a screenshot for this one from our previous version of the solution. The functionality of this playbook is unchanged, it's only the integration account reference we're updating so it's easier/possible for a deploying user to re-use an integration account across the playbooks.

https://github.com/tanium/Azure-Sentinel/blob/tanium-solution-2.0.1/Solutions/Tanium/Playbooks/Tanium-QuarantineHosts/images/Tanium-QuarantineHosts.png

File: Solutions/Tanium/Playbooks/Tanium-ComplyFindings/azuredeploy.json
Comment (sdball) [reply Solutions/Tanium/Playbooks/Tanium-ComplyFindings/azuredeploy.json:1]: How are you deploying the template? When I take this azuredeploy.json file to the custom deployment in Azure Portal I can deploy the logic app without issue: https://portal.azure.com/#create/Microsoft.Template

image

image

File: Solutions/Tanium/Playbooks/Tanium-ResolveThreatResponseAlert/azuredeploy.json
Comment (sdball) [reply Solutions/Tanium/Playbooks/Tanium-ResolveThreatResponseAlert/azuredeploy.json:1]: Whoop, sorry about that. I've pushed up an updated version of this playbook and rebuilt the 2.0.1 package zip.

File: Solutions/Tanium/Playbooks/Tanium-ComplyFindings/azuredeploy.json
Comment (v-vdixit) [reply Solutions/Tanium/Playbooks/Tanium-ComplyFindings/azuredeploy.json:1]: Thanks for this update @sdball, we are deploying this from the solution package contents and getting deployment error, thanks!

File: Solutions/Tanium/Playbooks/Tanium-ResolveThreatResponseAlert/azuredeploy.json
Comment (v-vdixit) [reply Solutions/Tanium/Playbooks/Tanium-ResolveThreatResponseAlert/azuredeploy.json:1]: Thanks for the update @sdball we will check on the changes and provide you update by 26 Jul 2023

File: Solutions/Tanium/Playbooks/Tanium-ComplyFindings/azuredeploy.json
Comment (sdball) [reply Solutions/Tanium/Playbooks/Tanium-ComplyFindings/azuredeploy.json:1]: @v-vdixit I setup a new empty resource group and log analytics workspace connected to Sentinel and imported my entire `mainTemplate.json` file and then deployed ComplyFindings from the playbook template and that also worked fine for me.

image

File: Solutions/Tanium/Playbooks/Tanium-ResolveThreatResponseAlert/azuredeploy.json
Comment (v-atulyadav) [reply Solutions/Tanium/Playbooks/Tanium-ResolveThreatResponseAlert/azuredeploy.json:1]: Hi @sdball, we have still facing deployment issue for this playbook, can you please attach screenshot of successful working of this playbook. Thanks

**Comment by v-atulyadav** [comment]
Hi @sdball, we are experiencing the deployment problem for other playbooks as well. Thanks for the screenshot. Could we schedule a call to discuss this issue? My email address is v-atulyadav@microsoft.com and we work in IST time. Thanks

**Comment by sdball** [comment]
@v-atulyadav I've sent you an email from my tanium.com email address to coordinate further.

**Comment by v-atulyadav** [comment]
ok @sdball will check once by deploying maintemplate from older zip and if same issue persist then will share error screenshot. Thanks

**Comment by sdball** [comment]
@v-atulyadav The older zip is the one with the issue. Please ensure you use the current zip from 269bcc30448cf66270776dffd1b695314bfdbab3 for testing.

**Comment by sdball** [comment]
To be certain you can check the SHA256 of the `2.0.1.zip` file and it should match this version:

```
aa8e119d43aea4146eeaf9f91f69c37db74f72683c0f09527b10b994f41041d4  2.0.1.zip
```

**Comment by v-atulyadav** [comment]
sure @sdball. Thanks

**Comment by v-atulyadav** [comment]
Hi @sdball, we are investigating this and will update you after we discuss a few things internally. Thanks

**Comment by v-atulyadav** [comment]
Hi @sdball,
I have tried again, but I am still experiencing the same problem. The maintemplate has been deployed successfully from the commit you mentioned in the PR, but when I attempt to create the playbook, an integration account error occurs. Please refer to the screenshot below. Also, we have tried giving the test integration parameter value.
image

**Comment by sdball** [comment]
@v-atulyadav Oh great! This error is actually a different one than you encountered previously. Before you were getting the error

```
The template parameter `IntegrationAccountPricingTier` is not found.
```

The error you are encountering here

```
The referenced integration account … could not be found.
```

That error is expected. The logic apps must be supplied with a valid Azure integration account that you have access to. The values `LogicAppsJavaScript` and `RG_1` reference our development integration account that you would not have access to.

You must create an [Azure integration account](https://portal.azure.com/#view/HubsExtension/BrowseResource/resourceType/Microsoft.Logic%2FintegrationAccounts) in advance before being able to successfully deploy these playbooks.

**Comment by v-atulyadav** [comment]
Hi @sdball, 
The playbooks have been successfully created after the integration account was created according to your instructions. However, we do not have API keys. Could you please share screenshots of successful running playbooks with us so we can proceed with merging?
Thanks.

**Comment by sdball** [comment]
@v-atulyadav Certainly!

Tanium-Apply-SecurityPatches-WithoutReboot
Tanium-Apply-SecurityPatches-WithoutReboot

Tanium-Apply-SecurityPatches-WithReboot
Tanium-Apply-SecurityPatches-WithReboot

Tanium-ComplyFindings
Tanium-ComplyFindings

Tanium-GeneralHostInfo
Tanium-GeneralHostInfo

Tanium-MSDefenderHealth
Tanium-MSDefenderHealth

Tanium-QuarantineHosts
Tanium-QuarantineHosts

Tanium-ResolveThreatResponseAlert
Tanium-ResolveThreatResponseAlert

Tanium-SCCMClientHealth
Tanium-SCCMClientHealth

Tanium-SecurityPatches
Tanium-SecurityPatches

Tanium-UnquarantineHosts
Tanium-UnquarantineHosts

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**